backends-common/postgres/src/main/java/org/apache/james/backends/postgres/utils/PostgresExecutor.java

```diff
@@ -145,8 +143,6 @@ public Flux<Record> executeDeleteAndReturnList(Function<DSLContext, DeleteResult
                     .flatMapMany(queryFunction)
                     .timeout(postgresConfiguration.getJooqReactiveTimeout())
                     .doOnError(TimeoutException.class, e -> LOGGER.error(JOOQ_TIMEOUT_ERROR_LOG, e))
-                    .collectList()
```

Comment:
The hanging issue occurs when connection-A is processing a query, and another where of the code tries to use connection-A for a different query. 

I guess this issue might be resolved when we using a connection pool (implemented)
Wait to CI result for confirmation